feature-libs/product/configuration/src/textfield-configurator/facade/configurator-textfield.service.ts

```diff
@@ -66,32 +73,48 @@ export class ConfiguratorTextfieldService {
    * Adds the textfield configuration to the cart, specified by its product code.
    *
    * @param productCode - Product code
+   * @param configuration Textfield configuration
    */
-  public addToCart(productCode: string) {
+  addToCart(
+    productCode: string,
+    configuration: ConfiguratorTextfield.Configuration
+  ) {
     this.activeCartService.requireLoadedCart().subscribe((cartState) => {
       const addToCartParameters: ConfiguratorTextfield.AddToCartParameters = {
         userId: this.configuratorUtils.getUserId(cartState.value),
         cartId: this.configuratorUtils.getCartId(cartState.value),
         productCode: productCode,
+        configuration: configuration,
         quantity: 1,
       };
-      this.callAddToCartActionWithConfigurationData(addToCartParameters);
+      this.store.dispatch(
+        new ConfiguratorTextfieldActions.AddToCart(addToCartParameters)
+      );
     });
   }
 
   /**
    * Updates a cart entry, specified by its cart entry number.
    *
    * @param cartEntryNumber - Cart entry number
+   * @param configuration Textfield configuration (list of alphanumeric attributes)
    */
-  public updateCartEntry(cartEntryNumber: string) {
+  public updateCartEntry(
+    cartEntryNumber: string,
+    configuration: ConfiguratorTextfield.Configuration
+  ) {
     this.activeCartService.requireLoadedCart().subscribe((cartState) => {
       const updateCartParameters: ConfiguratorTextfield.UpdateCartEntryParameters = {
         userId: this.configuratorUtils.getUserId(cartState.value),
         cartId: this.configuratorUtils.getCartId(cartState.value),
         cartEntryNumber: cartEntryNumber,
+        configuration: configuration,
       };
-      this.callUpdateCartEntryActionWithConfigurationData(updateCartParameters);
+      this.store.dispatch(
+        new ConfiguratorTextfieldActions.UpdateCartEntryConfiguration(
+          updateCartParameters
+        )
+      );
     });
   }
 
```

Comment:
The same comment for `public`, return type and refactor applies to `readConfigurationForCartEntry`